rules/prefer-type-error.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+const tcIdentifiers = new Set([
+	'isArguments',
+	'isArray',
+	'isArrayBuffer',
+	'isArrayLike',
+	'isArrayLikeObject',
+	'isBoolean',
+	'isBuffer',
+	'isDate',
+	'isElement',
+	'isEmptyObject',
+	'isError',
+	'isFinite',
```

Comment:
It just looks odd if you are forced to write this

```js
if (!Number.isFinite(x)) {
    throw new TypeError('The number is not finite'));
}
```

They might be interpreted like a special type, users don't see it like that. So I agree, it's a type error to JavaScript (technically), but not to developers. It's counter-intuitive.